src/server/game/Maps/Map.cpp

```diff
@@ -2989,20 +2990,14 @@ bool InstanceMap::CanEnter(Player* player)
                 if (iPlayer->IsGameMaster()) // bypass GMs
                     continue;
                 if (!player->GetGroup()) // player has not group and there is someone inside, deny entry
-                {
-                    player->SendTransferAborted(GetId(), TRANSFER_ABORT_MAX_PLAYERS);
-                    return false;
-                }
+                    return CANNOT_ENTER_MAX_PLAYERS;
                 // player inside instance has no group or his groups is different to entering player's one, deny entry
                 if (!iPlayer->GetGroup() || iPlayer->GetGroup() != player->GetGroup())
```

Comment:
Couldn't this be replaced with if (player->GetGroup() || (!iPlayer->GetGroup() || iPlayer->GetGroup() != player->GetGroup()) to be more efficient ?